File: EAGetMail/Samples_NetDesktop/CSharp/ImapFull/AppProject1/FormAddFolder.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace AppProject1
{
    public partial class FormAddFolder : Form
    {
        public FormAddFolder()
        {
            InitializeComponent();
        }

        private void ButtonOK_Click(object sender, EventArgs e)
        {
            textFolderName.Text = textFolderName.Text.Trim();
            if (textFolderName.Text.Length == 0)
            {
                MessageBox.Show("Please input folder name!");
                this.DialogResult = DialogResult.None;
            }
            else
            {
                this.DialogResult = DialogResult.OK;
            }
        }
    }
}

File: EAGetMail/Samples_NetDesktop/VB/ImapFull/AppProject1/FormAddFolder.vb
Public Class FormAddFolder
    Private Sub ButtonOK_Click(sender As Object, e As EventArgs) Handles ButtonOK.Click
        textFolderName.Text = textFolderName.Text.Trim()

        If textFolderName.Text.Length = 0 Then
            MessageBox.Show("Please input folder name!")
            Me.DialogResult = DialogResult.None
        Else
            Me.DialogResult = DialogResult.OK
        End If
    End Sub
End Class